runtime/src/bank.rs

```diff
@@ -3374,7 +3374,8 @@ impl Bank {
                     let mut account = self
                         .get_account_with_fixed_root(&pubkey)
                         .unwrap_or_default();
-                    account.lamports += rent_to_be_paid;
+                    // unwrap since no straightforward ability to do something useful with error
+                    account.checked_add_lamports(rent_to_be_paid).unwrap();
```

Comment:
Why not `saturating_add()`?  I feel it's better to drop rent payments at u64::MAX than crash the cluster.  Somewhat  philosophical I guess since this'll surely never happen either way

Reply:
(that is, crash the cluster *here*.  I suspect we'd trigger a capitalization assert later if we didn't crash here)

Reply:
@mvines suggests if add fails, then transfer rent_to_be_paid to: incinerator::id()

Reply:
> @mvines suggests if add fails, then transfer rent_to_be_paid to: incinerator::id()

Let's add metrics and logging along with that :fire:

Reply:
@t-nelson can you tell me what you would like logged? influx? info!? What do you want the log point called? I bet you have an idea of what you're looking for, and I sense a high likelihood my first attempt will not be sufficient.

Reply:
How about something like
```
error!("Incinerated {} rent lamports", rent_to_be_paid);
inc_new_counter_error!("bank-incinerated_rent_lamports", rent_to_be_paid);